### PR TITLE
Version 6 documentation

### PR DESCRIPTION
**Pull request by linuswillner**
This is the final documentation for version 6. It was migrated from TheSharks/WildBeastDocs in order to make maintaining things easier in future.

Main changes:
- The entire documentation has received an overhaul to accommodate for v6.
- Docs-related files like MkDocs dependencies and configuration have been migrated to this repository.
- Contribution guidelines, CODEOWNERS and other GitHub metadata has been merged. The original CONTRIBUTING.md was recasted as SOURCE_CONTRIBUTING.md and the docs contribution guidelines were named DOCS_CONTRIBUTING.md to distinguish between the two.
- Added docs runner script to NPM package.

This addresses the last blocking issue I detailed in #322 and thus clears WildBeast version 6 for release. As discussed with @Dougley today, we will perform CI and CNAME migrations on 8 June before releasing version 6.

### CODE REVIEW

File: .github/SOURCE_CONTRIBUTING.md [hunk 1]
```diff
@@ -1,36 +1,47 @@
...
 Please follow these rules when making contributions to this repository.
 
 # Contribution rules
+
 ## Unwanted contributions
+
 1. Style violations fixes
 2. ESLint rules changes
 3. Alias additions (Only applies if no new commands are provided)
-4. Issues stemming from usage of normal user accounts instead of OAuth bot accounts
```
Comment (Dougley) [inline .github/SOURCE_CONTRIBUTING.md]: Why is this removed?

File: docs/.circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+version: 2
```
Comment (Dougley) [inline docs/.circleci/config.yml]: This file isn't really needed, mkdocs doesn't include arbitrary files in the final build anyway

File: .github/SOURCE_CONTRIBUTING.md (hunk 1 above)
Comment (linuswillner) [reply .github/SOURCE_CONTRIBUTING.md]: We dropped support for normal user accounts in commit 6ac89ec on April 30, 2017.

File: docs/.circleci/config.yml (hunk 2 above)
Comment (linuswillner) [reply docs/.circleci/config.yml]: Thought so, I had listed as a TODO to remove it but kept it around just in case. Will remove it.

File: .github/SOURCE_CONTRIBUTING.md (hunk 1 above)
Comment (Dougley) [reply .github/SOURCE_CONTRIBUTING.md]: From that commit we dropped the ability to use email/password to login, but that doesn't take away that people could still use user account tokens to connect.

Comment (linuswillner) [reply .github/SOURCE_CONTRIBUTING.md]: True. However, I introduced this line myself in 2016 in commit acebebc, namely referring to the usage of the user and password login feature, a feature that was dropped in 6ac89ec.

As for whether it should stay around or not, I think that in the current Discord ecosystem not running bots under a normal user account is expected, and thus it is not necessary for us to state that explicitly.

Comment (linuswillner) [reply .github/SOURCE_CONTRIBUTING.md]: Then isn't that warning sufficient enough? I don't see why that should be in the contribution guidelines in the first place either (Which is on me, I fully acknowledge I'm part of the problem lol) - if anything it concerns end users. The line was not in any sort of issue template, and on Discord it's not really necessary to state it as previously mentioned.

**Comment by Dougley** [comment]
Renaming the contribution guidelines might result in GitHub not recognizing them properly by the way. Current behavior makes GitHub show a warning about the guidelines

**Comment by linuswillner** [comment]
Oh right... Hmm, should I rename the source contribution guidelines to CONTRIBUTING? Or should I merge the two? What do you think?

**Comment by Dougley** [comment]
Merging them would be better

**Comment by linuswillner** [comment]
Merged the guidelines.
